controllers/trainingjob/trainingjob_controller.go

```diff
@@ -185,8 +191,24 @@ func (r *Reconciler) reconcileTrainingJob(ctx reconcileRequestContext) error {
 		break
 
 	case sagemaker.TrainingJobStatusCompleted:
-		if err = r.addModelPathToStatus(ctx); err != nil {
-			return r.updateStatusAndReturnError(ctx, ReconcilingTrainingJobStatus, "", errors.Wrap(err, "Unable to add model path to status"))
+		if len(ctx.TrainingJob.Status.ModelPath) == 0 {
```

Comment:
Thanks for adding this optimization. I think we need to do more than check if the string is empty, though. We should strive to support full eventual consistency, such that if the model path is incorrect we update it. 

We can either update it every time (as it was before) or modify `addModelPathToStatus` so that it only updates if the status model path is different than the actual model path.

Reply:
I will remove the optimization for now and later worry about how to filter update if the path differs.

Reply:
Also when do u think mode path will be non-empty and in-correct ?

Reply:
I don't know when it will be incorrect, but defensive code is better for maintainability.